cpp/include/cudf/functions.h

```diff
@@ -944,3 +944,55 @@ gdf_error gdf_order_by(gdf_column** input_columns,
                        size_t       num_inputs,
                        gdf_column*  output_indices,
                        int          flag_nulls_are_smallest);
+
+
+/** 
+ * @brief Scatters the values of a source column into a destination column.
+ * 
+ * Scatters the elements of the source column into the destination column according
+ * to a scatter map such that source[i] will be scattered to destination[ scatter_map[i] ]
+ * 
+ * The number of elements in the scatter_map must equal the number of elements in the source
+ * column. 
+ * 
+ * If any index in scatter_map is outside the range of [0, destination->size), it 
+ * will be ignored.
+ * 
+ * If the same index appears more than once in scatter_map, the result is undefined.
+ * 
+ * @Param[in] source The input column whose rows will be scattered
+ * @Param[in] scatter_map An array that maps rows in the input column
+ * to rows in the output column. The size of the the scatter_map must equal
+ * the size of the source column. 
+ * @Param[out] destination A preallocated column equal in size to the source column that 
+ * will contain the rearrangement of the source column based on the mapping determined 
+ * by the scatter_map 
+ * 
+ * @Returns GDF_SUCCESS upon successful completion
+ */
+gdf_error gdf_scatter(gdf_column const * source, 
+                      gdf_column const * scatter_map, 
+                      gdf_column * destination);
+
+/**
+ * @brief Gathers the values from the source column into the destination column.
+ * 
+ * Gathers the elements of the source column into the destination column according 
+ * to a gather map such that destination[i] will contain source[ gather_map[i] ]
+ * 
+ * The number of elements in the gather_map must equal the number of elements in the 
+ * destination column. 
+ * 
+ * If gather_map[i] is outside the range [0, source->size),
+ * it will be ignored and the value of destination[i] will not be modified.
```

Comment:
Is there a reason to ignore it as opposed to throwing an out-of-bounds type error?